src/coreclr/jit/forwardsub.cpp

```diff
@@ -0,0 +1,549 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "jitpch.h"
+
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+// Simple Forward Substitution
+
+//------------------------------------------------------------------------
+// fgForwardSub: try forward substitution in this method
+//
+// Returns:
+//   suitable phase status
+//
+PhaseStatus Compiler::fgForwardSub()
+{
+    bool changed = false;
+    bool enabled = true;
+
+#if defined(DEBUG)
+    enabled = JitConfig.JitNoForwardSub() == 0;
+#endif
+
+    if (enabled)
+    {
+        for (BasicBlock* const block : Blocks())
+        {
+            JITDUMP("\n\n===> " FMT_BB "\n", block->bbNum);
```

Comment:
Is it worth skipping cold blocks here?

Reply:
Seems like this is the kind of thing that is not hot/cold sensitive?